flake.nix

```diff
@@ -0,0 +1,199 @@
+{
+  description = "Decrypt and encrypt agenix secrets inside Emacs";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/release-22.11";
+
+    bash-strict-mode = {
+      url = "github:sellout/bash-strict-mode";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+
+    flake-utils.url = "github:numtide/flake-utils";
+
+    homeManager = {
+      url = "github:nix-community/home-manager/release-22.11";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = inputs:
+    {
+      overlays = {
+        default = final: prev: {
+          emacsPackagesFor = emacs:
+            (prev.emacsPackagesFor emacs).overrideScope'
+            (inputs.self.overlays.emacs final prev);
+        };
+
+        emacs = final: prev: efinal: eprev: {
+          agenix = inputs.self.packages.${final.system}.agenix-el;
+        };
+      };
+
+      homeConfigurations.example = inputs.homeManager.lib.homeManagerConfiguration {
+        pkgs = import inputs.nixpkgs {
+          system = inputs.flake-utils.lib.system.aarch64-darwin;
```

Comment:
Could we move it inside `perSystem` and test in the CI

Reply:
There is an issue for having garnix run these (garnix-io/issues#24), but `homeConfigurations` is like `nixosConfigurations` in that it doesn’t expect a system component in the attribute path. It usually makes sense – you have a configuration for a specific machine, so the system is fixed. But in this case, it would be nice. And the way that flake-utils inserts the `system` into the tree is a bit too magical to just move this there.

We could do something like
```nix
{
  homeConfigurations =
    builtins.listToAddrs
    (builtins.map (system: {
                    name = "${system}-example";
                    value = inputs.homeManager.lib.homeManagerConfiguration ...;
                  })
                  inputs.flake-utils.defaultSystems);
}
```
which I wouldn’t be opposed to.

Reply:
OK, there is no need to rush until garnix supports these